File: aruco_detect_video.py
import cv2

dictionary = cv2.aruco.getPredefinedDictionary(cv2.aruco.DICT_6X6_250)

video = cv2.VideoCapture(0)

while True:
    ret, image = video.read()
    if not ret:
        break
    imagecopy = image.copy()
    corners, ids, _ = cv2.aruco.detectMarkers(image, dictionary)
    if ids is not None:
        cv2.aruco.drawDetectedMarkers(imagecopy, corners, ids)
    cv2.imshow("image", imagecopy)
    if cv2.waitKey(1) == ord('q'):
        break
cv2.destroyAllWindows()

File: aruco_detect_video.cpp
// https://docs.opencv.org/3.4/db/da9/tutorial_aruco_board_detection.html

#include <opencv2/aruco.hpp>
#include <opencv2/opencv.hpp>


int main() {
    cv::VideoCapture inputVideo;
    inputVideo.open(0);
    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
    while (inputVideo.grab()) {
        cv::Mat image, imageCopy;
        inputVideo.retrieve(image);
        image.copyTo(imageCopy);
        std::vector<int> ids;
        std::vector<std::vector<cv::Point2f> > corners;
        cv::aruco::detectMarkers(image, dictionary, corners, ids);
        // if at least one marker detected
        if (ids.size() > 0)
            cv::aruco::drawDetectedMarkers(imageCopy, corners, ids);
        cv::imshow("out", imageCopy);
        char key = (char)cv::waitKey(1);
        if (key == 27)
            break;
    }
}